include/git2/checkout.h

```diff
@@ -313,6 +313,12 @@ GIT_EXTERN(int) git_checkout_init_options(
  * Updates files in the index and the working tree to match the content of
  * the commit pointed at by HEAD.
  *
+ * Note that this is _not_ the correct mechanism used to switch branches;
+ * do not change your `HEAD` and then call this method, that would leave
+ * you with checkout conflicts since your working directory would then
+ * appear to be dirty.  Instead, checkout the target of the branch and
+ * then update `HEAD` if that was successful.
```

Comment:
Should we mention `git_repository_set_head()` here, as that's the convenient way we offer to switch the active branch.